pdns/nameserver.cc

```diff
@@ -89,6 +92,36 @@ extern StatBag S;
 vector<ComboAddress> g_localaddresses; // not static, our unit tests need to poke this
 
 void UDPNameserver::bindAddresses()
+{
+#ifdef HAVE_SYSTEMD
+	if (sd_listen_fds(0) == 0) {
+		bindLocalAddresses();
+	} else {
+		listenSystemdAddresses();
+	}
+#else
+	bindLocalAddresses();
+#endif
+}
+
+void UDPNameserver::listenSystemdAddresses()
+{
+	for (int i = 0; i <  sd_listen_fds(0); i++) {
```

Comment:
```suggestion
	for (int i = 0; i < sd_listen_fds(0); i++) {
```
(same for `TCPNameserver`)